src/PerfView/EventViewer/EventWindow.xaml.cs

```diff
@@ -1154,18 +1154,18 @@ public void Update()
             }
             else
             {
-                var match = Regex.Match(StartTextBox.Text, @"^\s*([\d\.,]+)\s+([\d\.,]+)\s*$");
-                if (match.Success)
+                if (RangeUtilities.TryParse(StartTextBox.Text, out var startTimeRelativeMSec, out var endTimeRelativeMSec))
                 {
-                    m_source.StartTimeRelativeMSec = 0;
-                    double.TryParse(match.Groups[1].Value, out m_source.StartTimeRelativeMSec);
-
-                    m_source.EndTimeRelativeMSec = m_source.MaxEventTimeRelativeMsec;
-                    double.TryParse(match.Groups[2].Value, out m_source.EndTimeRelativeMSec);
+                    m_source.StartTimeRelativeMSec = startTimeRelativeMSec;
+                    m_source.EndTimeRelativeMSec = endTimeRelativeMSec;
                 }
                 else
                 {
-                    if (!double.TryParse(StartTextBox.Text, out m_source.StartTimeRelativeMSec))
+                    if (double.TryParse(StartTextBox.Text, out startTimeRelativeMSec))
+                    {
+                        m_source.StartTimeRelativeMSec = startTimeRelativeMSec;
```

Comment:
I think you will want to keep the old behavior here in this else clause.  With your change here, if the value can't be parsed into a double, then the existing value of the field will be whatever the user just typed.  Without the change, it will revert back to 0, which is what we want.